builder/actions/cmake.py

```diff
@@ -2,14 +2,27 @@
 # SPDX-License-Identifier: Apache-2.0.
 
 import argparse
+from functools import lru_cache
 import os
 from pathlib import Path
+import shutil
 
 from builder.core.action import Action
 from builder.core.toolchain import Toolchain
 from builder.core.util import UniqueList
 
 
+@lru_cache(1)
```

Comment:
What is this MAGIC?

Reply:
it caches the results of the function
so we only end up looking for cmake once